src/linuxthreads.cc

```diff
@@ -365,15 +379,25 @@ static void ListerThread(struct ListerParams *args) {
             pid = local_atoi(ptr);
 
             /* Attach (and suspend) all threads                              */
-            if (pid && pid != clone_pid) {
+            if (pid) {
               struct kernel_stat tmp_sb;
               char fname[entry->d_reclen + 48];
-              strcat(strcat(strcpy(fname, "/proc/"), entry->d_name), marker_path);
+
+              /* If pid belong to the one called the library no need to proceed */
+              if (expid == 0 && pid == clone_pid) continue;
+
+              strcat(strcpy(fname, "/proc/"), entry->d_name);
```

Comment:
no strcat or strcpy please

Reply:
The original code is using strcat/strcpy, if you insist I can add a local alternate for strcat or cpy.

Reply:
it's fine